### DISCUSSION TITLE
When no sub-pages empty sidebar

### DISCUSSION THREAD

**Discussion by facelessuser** [squidfunk/mkdocs-material#2272] [Q&A]
<img width="1210" alt="Screen Shot 2021-02-04 at 7 57 28 AM" src="https://user-images.githubusercontent.com/1055125/106910908-e95df380-66be-11eb-9284-24de8efa04f3.png">

It used to be that when you were using tabs and there were no sub-pages, it would at least show the section IIRC, at the very least it used to show something. Now you just get a giant dead space, which looks weird. Or maybe I'm not used to it.

I think this started happening when you reworked things for the section feature (which obviously is not available on the standard Material). Was this intentional? I'm not sure why, but this is driving me crazy.

**Comment by facelessuser** [comment]
Yeah, it seems the section does not show if you do the following:

```yml
- section: file.md
```

But it does show if you do something like this:

```yml
- section:
   - section: file.md
```

I feel like the current section should show in both cases. Am I alone in this thought?

**Reply by facelessuser** [reply]
I guess to force a section, if you don't want the redundant link, you can just do something like:

```yml
- section:
    - '': file.md
```

I guess I can kind of see the logic, if the section links to one page, no sense showing the section breakdown. I guess after seeing this for so long in Material always showing the section, the change just bothers me.

I guess the workarounds above allows you to get old behavior and even tweak what shows (just section or section and link), but I'm not sure what is more correct. Should we not show anything at all in this case, is that the right default approach?

**Comment by facelessuser** [comment]
I guess an argument for showing something is that the link in the sidebar allows you to jump to the top of the page quite quickly 🤷

**Reply by squidfunk** [reply]
Related explanation: https://github.com/squidfunk/mkdocs-material/issues/2125#issuecomment-744360768

Yes, the behavior is different and I'm undecided what we should do (or if we should do something). The grouping semantics are different now, as they now allow for top-level links to be added to the tabs navigation. Before, all top-level pages were grouped under the first tab. Now, as there are no sub-pages, there's nothing to render, except for the title, which wouldn't be clickable anyway, so there would be no go to top link.

The linked comment outlines how to (nearly) get the old behavior. I'm open to discussing if it makes sense to show something, given that we're not changing the grouping semantics for tabs or sections navigation.

> I think this started happening when you reworked things for the section feature (which obviously is not available on the standard Material). Was this intentional? I'm not sure why, but this is driving me crazy.

Only sticky tabs and section index pages are solely available to Insiders. Everything else was released when hitting the latest funding goal.

**Reply by facelessuser** [reply]
Understood. I hadn't realized this change until I looked at my docs recently. I can mimic the old way if it bothers me enough I guess. I just wasn't sure if it was known/intended or not. At least I now know how to work around if I can't simply accept it 🙂

**Reply by squidfunk** [reply]
As said, I'm open for discussion 😊 It was an error to introduce this change as part of a minor version, it should have been a major release, but it felt wrong fixing a bug and bumping a major version (again). However, that would have communicated the intent more clearly.
